include/glow/ExecutionEngine/Context.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_EXECUTIONENGINE_CONTEXT_H
+#define GLOW_EXECUTIONENGINE_CONTEXT_H
+
+#include "glow/Base/Tensor.h"
+
+#include <unordered_map>
+
+namespace glow {
+
+class Tensor;
+class Storage;
+class Placeholder;
+
+/// This class provides a mapping between some graph nodes, which are a symbolic
+/// representation of some computation, and concrete tensors that represent the
+/// inputs and outputs to the graph. The context owns the tensors and the graph
+/// uses these values as runtime. This is useful for the multi-threaded
+/// execution of code, where each thread has a different execution context. The
+/// difference between this class and a regular map is that the Context owns the
+/// Tensors (not only the pointers) and manages their lifetime.
+class Context final {
+  /// Maps Placeholders to Tensors.
+  std::unordered_map<Placeholder *, Tensor *> map_;
```

Comment:
Should this be a PlaceholderMap?

Reply:
That seems very reasonable.  Passing a 'Context' to the various compile()'s seems a little "nicer".